engine/src/main/java/org/terasology/logic/players/LocalPlayerSystem.java

```diff
@@ -212,6 +215,33 @@ private void standPlayer(EntityRef entity) {
         clientComp.character.send(new SetMovementModeEvent(MovementMode.WALKING));
     }
 
+    /**
+     * Auto move is disabled every time any other movement event occurs.
+     */
+    private void stopAutoMove() {
+        isAutoMove = false;
+    }
+
+    /**
+     * Start a thread that moves the entity in the camera direction at every period of 0.2 seconds.
+     * @param entity
+     */
+    private void startAutoMove(EntityRef entity) {
+        isAutoMove = true;
+        ClientComponent clientComponent = entity.getComponent(ClientComponent.class);
+        new Thread(() -> {
+            while(isAutoMove) {
+                try {
+                    Vector3f viewDir = entity.getComponent(LocationComponent.class).getWorldDirection();
+                    clientComponent.character.send(new CharacterImpulseEvent(viewDir.mul(8f)));
```

Comment:
@skaldarnar : What can I use to make it feel smooth? I was looking for something that could simulate a key press event with the Forwards key pressed at all times but that _drops_ inputs because of too many inputs from the entity.